### PR TITLE
Upgrade to typescript 4.2.2 everywhere

### PR DESCRIPTION
**Pull request by xianny**
## Description

Following https://github.com/0xProject/tools/pull/28; it needs typescript v4  to work. Figured we were overdue anyway. 

## Checklist:

-   [ ] Prefix PR title with `[WIP]` if necessary.
-   [ ] Add tests to cover changes as needed.
-   [ ] Update documentation as needed.
-   [ ] Add new entries to the relevant CHANGELOG.jsons.

### CODE REVIEW

File: web3-wrapper/src/web3_wrapper.ts [hunk 1]
```diff
@@ -715,19 +714,27 @@ export class Web3Wrapper {
...
+        // tslint:disable:no-object-literal-type-assertion
         const payloadWithDefaults = {
             id: this._jsonRpcRequestId++,
             params: [],
             jsonrpc: '2.0',
             ...payload,
-        };
-        const response = await promisify<JSONRPCResponsePayload>(sendAsync)(payloadWithDefaults);
-        if (response.error) {
-            throw new Error(response.error.message);
```
Comment (dorothy-zbornak) [inline web3-wrapper/src/web3_wrapper.ts:727]: Don't we still want to throw on this?

**Review by dorothy-zbornak — APPROVED** [review]
tytyty!
I know this was annoying because I've tried before.
One Q then :+1:

File: web3-wrapper/src/web3_wrapper.ts (hunk 1 above)
Comment (xianny) [reply web3-wrapper/src/web3_wrapper.ts:727]: I kinda tested this, it looks like `await sendAsync` will throw, like a normal `await {PromiseReturningFunction}` when the promise rejects. 

Looking at this code again though I think the error handling doesn't make sense, will update. 🙏

File: web3-wrapper/src/web3_wrapper.ts [hunk 2]
```diff
@@ -715,19 +714,23 @@ export class Web3Wrapper {
...
+        } as JSONRPCRequestPayload;
+        // tslint:enable:no-object-literal-type-assertion
+        const sendAsync = promisify(this._provider.sendAsync.bind(this._provider));
+        const response = await sendAsync(payloadWithDefaults); // will throw if it fails
+        if (!response || !response.result) {
+            throw new Error(`No response`);
```
Comment (dorothy-zbornak) [inline web3-wrapper/src/web3_wrapper.ts:731]: I think we really want to throw the `response.error.message` if it's available to us.

**Pull request merged**